src/mpid/ch4/netmod/ofi/util.c

```diff
@@ -260,6 +260,32 @@ int MPIDI_OFI_control_handler(int handler_id, void *am_hdr,
 #define isDOUBLE_COMPLEX(x) ((x) == MPI_DOUBLE_COMPLEX || (x) == MPI_COMPLEX8 || \
                               (x) == MPI_C_DOUBLE_COMPLEX)
 
+#undef FUNCNAME
+#define FUNCNAME check_mpi_acc_valid
+#undef FCNAME
+#define FCNAME MPL_QUOTE(check_mpi_acc_valid)
+MPL_STATIC_INLINE_PREFIX bool check_mpi_acc_valid(MPI_Datatype dtype, MPI_Op op)
+{
+    bool valid_flag = FALSE;
+
+    /* Check if the <datatype, op> is supported by MPICH. Note that MPICH
+     * supports more combinations than that specified in standard (see definition
+     * of these checking routines for extended support). */
+
+    /* Predefined reduce operation, NO_OP, REPLACE */
+    if (op != MPI_OP_NULL) {
+        int mpi_errno;
+        mpi_errno = (*MPIR_OP_HDL_TO_DTYPE_FN(op)) (dtype);
+        if (mpi_errno == MPI_SUCCESS)
+            valid_flag = TRUE;
+    } else {
+        /* Compare and swap */
+        if (MPIR_Type_is_rma_atomic(dtype))
+            valid_flag = TRUE;
```

Comment:
`true`

Reply:
done. Also changed in ch4 layer for the new hints.